### DISCUSSION TITLE
rabbitmq_prometheus plugin can not listen to ipv6

### DISCUSSION THREAD

**Discussion by ruben-herold** [rabbitmq/rabbitmq-server#2672] [Other] [answered]
hi,

we have a nice running rabbitmq cluster on 3.8.9 with erlang 23.1.  The cluster is ipv6 only. 
I want enable the prometheus plugin for better monitoring.  But the plugin only listen on ipv4. 

I tried to add this lines to our config:

 {rabbitmq_prometheus,
 [
          {listener, [{port, 15692}, {ip, "::"}]}
 ]}
].

This also did not help.

**Comment by michaelklishin** [comment, marked as answer]
The plugin uses

``` ini
prometheus.tcp.ip = …
```

for listener interface configuration. If you absolutely must use the advanced config format, the key is [`rabbitmq_prometheus.tcp_config.ip`](https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbitmq_prometheus/priv/schema/rabbitmq_prometheus.schema#L31), not `rabbitmq_prometheus.listener.ip`.

It should be no different from any other HTTP-based listener in its capabilities.

**Comment by ruben-herold** [comment]
thx a lot. Can we improve the documentation?  I don't see the rabbitmq_prometheus.tcp_config.ip in any type of documentation I could found.

**Reply by michaelklishin** [reply]
The docs have been updated.
